Multiple entries with same key

### Stack trace

```
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Multiple entries with same key: LootItemConditionRandomChance[probability=0.25]=poke:effect_checking/movechance and LootItemConditionRandomChance[probability=0.25]=golem:random_lc
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[?:?]
	at net.minecraft.server.Main.main(Main.java:286) ~[paper-1.20.2.jar:git-Paper-197]
	at org.bukkit.craftbukkit.Main.main(Main.java:324) ~[paper-1.20.2.jar:git-Paper-197]
	at io.papermc.paperclip.Paperclip.lambda$main$0(Paperclip.java:42) ~[app:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IllegalArgumentException: Multiple entries with same key: LootItemConditionRandomChance[probability=0.25]=poke:effect_checking/movechance and LootItemConditionRandomChance[probability=0.25]=golem:random_lc
	at com.google.common.collect.ImmutableMap.conflictException(ImmutableMap.java:378) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:372) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:246) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.RegularImmutableMap.fromEntryArrayCheckingBucketOverflow(RegularImmutableMap.java:133) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:95) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:572) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.ImmutableMap$Builder.buildOrThrow(ImmutableMap.java:600) ~[guava-32.1.2-jre.jar:?]
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:587) ~[guava-32.1.2-jre.jar:?]
	at net.minecraft.world.level.storage.loot.LootDataManager.apply(LootDataManager.java:106) ~[paper-1.20.2.jar:git-Paper-197]
	at net.minecraft.world.level.storage.loot.LootDataManager.lambda$reload$2(LootDataManager.java:49) ~[paper-1.20.2.jar:git-Paper-197]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.server.packs.resources.SimpleReloadInstance.lambda$new$3(SimpleReloadInstance.java:66) ~[?:?]
	at net.minecraft.Util.blockUntilDone(Util.java:828) ~[?:?]
	at net.minecraft.Util.blockUntilDone(Util.java:817) ~[?:?]
	at net.minecraft.server.Main.main(Main.java:254) ~[paper-1.20.2.jar:git-Paper-197]
	... 3 more

```


### Plugin and Datapack List

[Golems 1.1.zip](https://github.com/PaperMC/Paper/files/12709088/Golems.1.1.zip)
[PokeMobs 1.8.zip](https://github.com/PaperMC/Paper/files/12709091/PokeMobs.1.8.zip)


### Actions to reproduce (if known)

_No response_

### Paper version

Paper 1.20.2-197
(Experimental Build 1 on https://papermc.io/downloads/paper)

Server couldn't start up so hope this suffices


### Other

_No response_

--- a/patches/server/1035-Fix-creating-the-reverse-map-for-loot-table-keys.patch
+++ b/patches/server/1035-Fix-creating-the-reverse-map-for-loot-table-keys.patch
@@ -1,0 +1,36 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Tue, 26 Sep 2023 18:32:49 -0700
+Subject: [PATCH] Fix creating the reverse map for loot table keys
+
+
+diff --git a/src/main/java/net/minecraft/world/level/storage/loot/LootDataManager.java b/src/main/java/net/minecraft/world/level/storage/loot/LootDataManager.java
+index dd375fffa727db76fa989248b9b836960974c372..3563fdb8f8d3dbaabc611ab5cdba1d771406b336 100644
+--- a/src/main/java/net/minecraft/world/level/storage/loot/LootDataManager.java
++++ b/src/main/java/net/minecraft/world/level/storage/loot/LootDataManager.java
+@@ -30,7 +30,7 @@ public class LootDataManager implements PreparableReloadListener, LootDataResolv
+     public static final LootDataId<LootTable> EMPTY_LOOT_TABLE_KEY = new LootDataId<>(LootDataType.TABLE, BuiltInLootTables.EMPTY);
+     private Map<LootDataId<?>, ?> elements = Map.of();
+     private Multimap<LootDataType<?>, ResourceLocation> typeKeys = ImmutableMultimap.of();
+-    public Map<?, ResourceLocation> lootTableToKey = ImmutableMap.of(); // CraftBukkit
++    public Map<LootTable, ResourceLocation> lootTableToKey = ImmutableMap.of(); // CraftBukkit // Paper - only store loot tables because those don't override equals and hashCode
+ 
+     public LootDataManager() {}
+ 
+@@ -101,8 +101,14 @@ public class LootDataManager implements PreparableReloadListener, LootDataResolv
+         this.elements = map1;
+         this.typeKeys = com_google_common_collect_immutablemultimap_builder.build();
+         // CraftBukkit start - build a reversed registry map
+-        ImmutableMap.Builder<Object, ResourceLocation> lootTableToKeyBuilder = ImmutableMap.builder();
+-        this.elements.forEach((key, lootTable) -> lootTableToKeyBuilder.put((Object) lootTable, key.location()));
++        // Paper start - only create a map for loot tables
++        ImmutableMap.Builder<LootTable, ResourceLocation> lootTableToKeyBuilder = ImmutableMap.builder();
++        this.elements.forEach((lootDataId, value) -> {
++            if (lootDataId.type() == LootDataType.TABLE && value instanceof LootTable lootTable) {
++                lootTableToKeyBuilder.put(lootTable, lootDataId.location());
++            }
++        });
++        // Paper end
+         this.lootTableToKey = lootTableToKeyBuilder.build();
+         // CraftBukkit end
+     }

